addons/sale_project/models/project_project.py

```diff
@@ -120,6 +123,16 @@ def action_customer_preview(self):
             'url': self.get_portal_url(),
         }
 
+    @api.onchange('reinvoiced_sale_order_id')
+    def on_change_reinvoiced_sale_order_id(self):
```

Comment:
No need to have that method in public. 🙂 
Also, by convention we name the onchange method like: `_onchange_{field_name_changed}`
```suggestion
    def _onchange_reinvoiced_sale_order_id(self):
```